flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java

```diff
@@ -170,15 +170,19 @@ public void execute(Context context) throws Exception {
                             LOG.info("Ignored {} record: {}", operationType, changeStreamDocument);
                     }
                 }
-
-                if (changeRecord != null) {
+                if (changeRecord != null && !isBoundedRead()) {
                     queue.enqueue(new DataChangeEvent(changeRecord));
                 }
 
                 if (isBoundedRead()) {
                     ChangeStreamOffset currentOffset;
                     if (changeRecord != null) {
                         currentOffset = new ChangeStreamOffset(getResumeToken(changeRecord));
+                        // The log after the high watermark won't emit.
+                        if (currentOffset.isAtOrBefore(streamSplit.getEndingOffset())) {
+                            queue.enqueue(new DataChangeEvent(changeRecord));
```

Comment:
Maybe `currentOffset.isAtOrBefore`  shoud be `currentOffset.isBefore`?

Reply:
Maybe in another PR? Now other cdc source read also includes high_watermark, this PR just fix bug, not to change the common logic?

Reply:
> Maybe in another PR? Now other cdc source read also includes high_watermark, this PR just fix bug, not to change the common logic?

Could it cause read same data twice if use includes high_watermark?

Reply:
This will fix in https://github.com/ververica/flink-cdc-connectors/issues/2885 for all connectors.But it still need to discuss more, for example, Whether to read skip high_watermark, or streaming phase skip low_watermark

Reply:
> This will fix in #2885 for all connectors.But it is a big influence, still need to discuss more, for example, Whether to read skip high_watermark, or streaming phase skip low_watermark.

ok